source/Calamari/Commands/Java/DeployJavaArchiveCommand.cs

```diff
@@ -77,7 +77,7 @@ public override int Execute(string[] commandLineArguments)
             var semaphore = SemaphoreFactory.Get();
             var journal = new DeploymentJournal(fileSystem, semaphore, variables);
             var jarTools = new JarTool(commandLineRunner, log, variables);
-            var packageExtractor = new JarPackageExtractor(jarTools);
+            var packageExtractor = new JarPackageExtractor(jarTools).WithExtractionLimits(log);
```

Comment:
sneaky custom package extractor 😄

Reply:
I know, so annoying. I think this is just forcing us to use the `Jar` extractor in this context regardless of the file extension, because if we used the `CombinedExtractor`'s precedence, then Java archives with a `zip` extension wouldn't be properly treated. I think.